static/js/components/widgets/PeopleSelector.js

```diff
@@ -0,0 +1,186 @@
+// @flow
+import React from "react"
+import R from "ramda"
+import Autosuggest from "react-autosuggest"
+import { arrayMove } from "react-sortable-hoc"
+import { connect } from "react-redux"
+
+import PeopleList from "./PeopleList"
+import { PeopleItem } from "./PeopleItem"
+import SearchTextbox from "../SearchTextbox"
+
+import { actions } from "../../actions"
+import { searchResultToProfile } from "../../lib/search"
+import { SEARCH_FILTER_PROFILE } from "../../lib/picker"
+
+import type { Dispatch } from "redux"
+import type { Profile } from "../../flow/discussionTypes"
+
+type Props = {
+  clearSearch: () => void,
+  profiles: Array<Profile>,
+  updateProfiles: (newJson: Array<Profile>) => void,
+  suggestions: Array<Profile>,
+  fetchSuggestions: (text: string) => Promise<Array<Profile>>
+}
+type State = {
+  text: string
+}
+export class PeopleSelector extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props)
+
+    this.state = {
+      text: ""
+    }
+  }
+
+  componentDidMount() {
+    const { clearSearch } = this.props
+    clearSearch()
+  }
+
+  onSuggestionsFetchRequested = ({
+    value,
+    reason
+  }: {
```

Comment:
This is a bit hard to read here, I had to read it a few times before I realized it was the type definition of the argument and not the return type. Might be worth extracting this to a top-level `type` definition for an easier read.

Reply:
Will do